ECS audit logging (#74640)
elastic/kibana: docs/settings/security-settings.asciidoc

============

|===

[float]
[[audit-logging-settings]]
===== Audit logging settings

You can enable audit logging to support compliance, accountability and security. When enabled Kibana will capture:

- Who performed an action
- What action was performed
- When it occurred

For more details and a reference of audit events see <<xpack-security-audit-logging>>.

[cols="2*<"]
|===
| `xpack.security.audit.enabled`
| Set to `true` to enable audit logging for security events. *Default:* `false`

> | `xpack.security.audit.ignore_filters[]` (beta)

Reviewer: These settings now are a bit of a blend between the legacy and new loggers. What do you think about introducing just the `xpack.security.audit.enabled` setting first, but then below get into the configuration for the new beta logger?

This would also let us add a heading to encapsulate all of the beta settings, where we can add a single `beta[]` indicator to warn users that everything in the section is a beta feature (instead of having `beta()` everywhere)

Author: That's a good idea, will do

----------------------------------------
<1> Filters out HTTP request events
<2> Filters out any data write events

[cols="2*<"]
|===
| `xpack.security.audit.ignore_filters[].actions[]` (beta)
| List of values matched against the `event.action` field of an audit event. See <<xpack-security-audit-logging>> for a list of available events.

| `xpack.security.audit.ignore_filters[].categories[]` (beta)
| List of values matched against the `event.category` field of an audit event. See https://www.elastic.co/guide/en/ecs/current/ecs-allowed-values-event-category.html[ECS categorization field] for allowed values.

| `xpack.security.audit.ignore_filters[].types[]` (beta)
| List of values matched against the `event.type` field of an audit event. See https://www.elastic.co/guide/en/ecs/current/ecs-allowed-values-event-type.html[ECS type field] for allowed values.

| `xpack.security.audit.ignore_filters[].outcomes[]` (beta)
| List of values matched against the `event.outcome` field of an audit event. See https://www.elastic.co/guide/en/ecs/current/ecs-allowed-values-event-outcome.html[ECS outcome field] for allowed values.

| `xpack.security.audit.appender` (beta)

> | Optional. Specifies where audit logs should be written to and how they should be formatted. *Default:* console

Reviewer: nit: We don't have a default set for this anymore, right?

```suggestion
| Optional. Specifies where audit logs should be written to and how they should be formatted.
```

2+a|
[NOTE]
============
If you set `xpack.security.audit.appender` the legacy audit logger will be disabled and the new audit logger will be used instead.
============
|===

For example:

[source,yaml]
----------------------------------------
xpack.security.audit.appender:
  kind: file
  path: /path/to/audit.log
  layout:
    kind: json
----------------------------------------

[cols="2*<"]
|===
| `xpack.security.audit.appender.kind` (beta)

> | Mandatory. Specifies where audit logs should be written to. Allowed values are `console` or `file` appender.

Reviewer: ```suggestion
| Mandatory. Specifies where audit logs should be written to. Allowed values are `console` or `file`.
```

|===

[float]

> [[file-appender-settings]]

Reviewer: Since this anchor is global to all of the kibana docs, we should probably make it more specific so that we don't collide with other anchors in the future

```suggestion
[[audit-logging-file-appender-settings]]
```

===== File appender settings (beta)

The file appender can be configured using the following settings:

[cols="2*<"]
|===
| `xpack.security.audit.appender.path` (beta)
| Mandatory. Full file path the log file should be written to.

| `xpack.security.audit.appender.layout.kind` (beta)
| Mandatory. Specifies how audit logs should be formatted. Allowed values are `json` or `pattern`.
|===

[float]

> [[pattern-layout-settings]]

Reviewer: Since this anchor is global to all of the kibana docs, we should probably make it more specific so that we don't collide with other anchors in the future

```suggestion
[[audit-logging-pattern-layout-settings]]
```

===== Pattern layout settings (beta)

The pattern layout can be configured using the following settings:

[cols="2*<"]
|===
| `xpack.security.audit.appender.layout.highlight` (beta)

> | Optional. Set to `true` to enable audit logging for security events

Reviewer: I don't think this description is correct?

Author: cmd + c / cmd + v 🤡

| `xpack.security.audit.appender.layout.pattern` (beta)

> | Optional. Specifies how the log line should be formatted. *Default:* `[%date][%level][%logger]%meta %message`

Reviewer: question: Are you aware of any docs we can link to which explains the different pattern variables that can be used?

Author: Unfortunately, I couldn't find the new logging settings documented in the customer facing docs. 
There's pretty good and extensive documentation in the readme file in `src/core/server/logging/README.md` but think that's it.
@restrry - Are these settings documented in the official docs too? If not are there plans to do that?

Reviewer 2: unfortunately, `src/core/server/logging/README.md` is the only document available at the moment. we are going to declare new logging system GA after https://github.com/elastic/kibana/issues/58261

============

|===

[float]
[[audit-logging-settings]]
===== Audit logging settings

You can enable audit logging to support compliance, accountability, and security. When enabled, {kib} will capture:

- Who performed an action
- What action was performed
- When the action occurred

For more details and a reference of audit events, refer to <<xpack-security-audit-logging>>.

[cols="2*<"]
|===
| `xpack.security.audit.enabled`
| Set to `true` to enable audit logging for security events. *Default:* `false`
|===

[float]
[[audit-logging-settings-beta]]
===== ECS audit logging settings

beta[]

To enable the <<xpack-security-ecs-audit-logging, ECS audit logger>>, specify where you want to write the audit events using `xpack.security.audit.appender`.

[cols="2*<"]
|===
| `xpack.security.audit.appender`

> | experimental[] Optional. Specifies where audit logs should be written to and how they should be formatted.

Reviewer: Sorry if there was already a conversation about this that I missed -- but we have the new settings described as "experimental" while the feature itself is "beta". Is there a reason for that discrepancy?

Author: Is there a difference between “beta” and “experimental”? Platform team suggested adding this in which made sense to me but if experimental is a stage before beta then that is probably not correct. Happy to remove the experimental flag

Reviewer: > Is there a difference between “beta” and “experimental”? 

Great question! Generally speaking, "experimental" features are something that we're exploring, but may change significantly or be removed entirely.

"Beta" features are perhaps less polished, and may still change, but we're more confident that that they'll stick around and become GA in the near future.

The definitions from our docs:

> This functionality is in beta and is subject to change. The design and code is less mature than official GA features and is being provided as-is with no warranties. Beta features are not subject to the support SLA of official GA features.

> This functionality is experimental and may be changed or removed completely in a future release. Elastic will take a best effort approach to fix any issues, but experimental features are not subject to the support SLA of official GA features.

Author: Cool, thanks for clarifying - I think this should be beta then rather than experimental! Will remove from the docs

| experimental[] Optional. Set to `true` to enable highlighting log messages with colors.

| `xpack.security.audit.appender.layout.pattern`
| experimental[] Optional. Specifies how the log line should be formatted. *Default:* `[%date][%level][%logger]%meta %message`
|===

[float]
[[audit-logging-ignore-filters]]
===== Ignore filters

[cols="2*<"]
|===
| `xpack.security.audit.ignore_filters[]`
| experimental[] List of filters that determine which events should be excluded from the audit log. An event will get filtered out if at least one of the provided filters matches.

2+a| For example:

[source,yaml]
----------------------------------------
xpack.security.audit.ignore_filters:
- actions: [http_request] <1>
- categories: [database]
  types: [creation, change, deletion] <2>
----------------------------------------
<1> Filters out HTTP request events
<2> Filters out any data write events

| `xpack.security.audit.ignore_filters[].actions[]`
| experimental[] List of values matched against the `event.action` field of an audit event. Refer to <<xpack-security-audit-logging>> for a list of available events.

| `xpack.security.audit.ignore_filters[].categories[]`

> | experimental[] List of values matched against the `event.category` field of an audit event. Refer to https://www.elastic.co/guide/en/ecs/current/ecs-allowed-values-event-category.html[ECS categorization field] for allowed values.

Reviewer: Should we link to the ECS version that we're claiming to support instead of `current`?

```suggestion
| experimental[] List of values matched against the `event.category` field of an audit event. Refer to https://www.elastic.co/guide/en/ecs/1.5/ecs-allowed-values-event-category.html[ECS categorization field] for allowed values.
```

| `xpack.security.audit.ignore_filters[].types[]`

> | experimental[] List of values matched against the `event.type` field of an audit event. Refer to https://www.elastic.co/guide/en/ecs/current/ecs-allowed-values-event-type.html[ECS type field] for allowed values.

Reviewer: ```suggestion
| experimental[] List of values matched against the `event.type` field of an audit event. Refer to https://www.elastic.co/guide/en/ecs/1.5/ecs-allowed-values-event-type.html[ECS type field] for allowed values.
```

| `xpack.security.audit.ignore_filters[].outcomes[]`

> | experimental[] List of values matched against the `event.outcome` field of an audit event. Refer to https://www.elastic.co/guide/en/ecs/current/ecs-allowed-values-event-outcome.html[ECS outcome field] for allowed values.

Reviewer: ```suggestion
| experimental[] List of values matched against the `event.outcome` field of an audit event. Refer to https://www.elastic.co/guide/en/ecs/1.5/ecs-allowed-values-event-outcome.html[ECS outcome field] for allowed values.
```

============

|===

[float]
[[audit-logging-settings]]
===== Audit logging settings

> You can enable audit logging to support compliance, accountability and security. When enabled {kib} will capture:

Reviewer 3: ```suggestion
You can enable audit logging to support compliance, accountability, and security. When enabled, {kib} will capture:
```

- Who performed an action
- What action was performed

> - When it occurred

Reviewer 3: ```suggestion
- When the action occurred
```

> For more details and a reference of audit events see <<xpack-security-audit-logging>>.

Reviewer 3: ```suggestion
For more details and a reference of audit events, refer to <<xpack-security-audit-logging>>.
```

Reviewer 3: To meet accessibility standards, we've removed instances of `see`, and replaced them with `refer to`.

[cols="2*<"]
|===
| `xpack.security.audit.enabled`
| Set to `true` to enable audit logging for security events. *Default:* `false`
|===

[float]
[[audit-logging-settings-beta]]

> ===== Audit logging settings (beta)

Reviewer 3: ```suggestion
===== Audit logging settings
```

Reviewer 3: I did a quick search in our docs for `beta`, and it looks like you only need `beta[]`, like you already have on line 265.

beta[]

> To enable the new audit logger (beta) you need to specify where the new audit

Reviewer 3: Could we shorten this? For example:

To enable the audit logger, specify where you want to write the audit events using `xpack.security.audit.appender`.

Author: Thanks for all your suggestions, looking much better now :)

events should be written to using the `xpack.security.audit.appender` option.

[cols="2*<"]
|===
| `xpack.security.audit.appender`
| Optional. Specifies where audit logs should be written to and how they should be formatted.

2+a| For example:

[source,yaml]
----------------------------------------
xpack.security.audit.appender:
  kind: file
  path: /path/to/audit.log
  layout:
    kind: json
----------------------------------------

| `xpack.security.audit.appender.kind`

> | Mandatory. Specifies where audit logs should be written to. Allowed values are `console` or `file`.

Reviewer 3: ```suggestion
| Required. Specifies where audit logs should be written to. Allowed values are `console` or `file`.
```

|===

[float]
[[audit-logging-file-appender]]
===== File appender

The file appender can be configured using the following settings:

[cols="2*<"]
|===
| `xpack.security.audit.appender.path`

> | Mandatory. Full file path the log file should be written to.

Reviewer 3: ```suggestion
| Required. Full file path the log file should be written to.
```

| `xpack.security.audit.appender.layout.kind`

> | Mandatory. Specifies how audit logs should be formatted. Allowed values are `json` or `pattern`.

Reviewer 3: ```suggestion
| Required. Specifies how audit logs should be formatted. Allowed values are `json` or `pattern`.
```

|===

[float]
[[audit-logging-pattern-layout]]
===== Pattern layout

The pattern layout can be configured using the following settings:

[cols="2*<"]
|===
| `xpack.security.audit.appender.layout.highlight`

> | Optional. Set to `true` to enable highlighting log messages with colors

Reviewer 3: ```suggestion
| Optional. Set to `true` to enable highlighting log messages with colors.
```

| `xpack.security.audit.appender.layout.pattern`
| Optional. Specifies how the log line should be formatted. *Default:* `[%date][%level][%logger]%meta %message`
|===

[float]
[[audit-logging-ignore-filters]]
===== Ignore filters

[cols="2*<"]
|===
| `xpack.security.audit.ignore_filters[]`
| List of filters that determine which events should be excluded from the audit log. An event will get filtered out if at least one of the provided filters matches.

2+a| For example:

[source,yaml]
----------------------------------------
xpack.security.audit.ignore_filters:
- actions: [http_request] <1>
- categories: [database]
  types: [creation, change, deletion] <2>
----------------------------------------
<1> Filters out HTTP request events
<2> Filters out any data write events

| `xpack.security.audit.ignore_filters[].actions[]`

> | List of values matched against the `event.action` field of an audit event. See <<xpack-security-audit-logging>> for a list of available events.

Reviewer 3: ```suggestion
| List of values matched against the `event.action` field of an audit event. Refer to <<xpack-security-audit-logging>> for a list of available events.
```

| `xpack.security.audit.ignore_filters[].categories[]`

> | List of values matched against the `event.category` field of an audit event. See https://www.elastic.co/guide/en/ecs/current/ecs-allowed-values-event-category.html[ECS categorization field] for allowed values.

Reviewer 3: ```suggestion
| List of values matched against the `event.category` field of an audit event. Refer to https://www.elastic.co/guide/en/ecs/current/ecs-allowed-values-event-category.html[ECS categorization field] for allowed values.
```

| `xpack.security.audit.ignore_filters[].types[]`

> | List of values matched against the `event.type` field of an audit event. See https://www.elastic.co/guide/en/ecs/current/ecs-allowed-values-event-type.html[ECS type field] for allowed values.

Reviewer 3: ```suggestion
| List of values matched against the `event.type` field of an audit event. Refer to https://www.elastic.co/guide/en/ecs/current/ecs-allowed-values-event-type.html[ECS type field] for allowed values.
```

| `xpack.security.audit.ignore_filters[].outcomes[]`

> | List of values matched against the `event.outcome` field of an audit event. See https://www.elastic.co/guide/en/ecs/current/ecs-allowed-values-event-outcome.html[ECS outcome field] for allowed values.

Reviewer 3: ```suggestion
| List of values matched against the `event.outcome` field of an audit event. Refer to https://www.elastic.co/guide/en/ecs/current/ecs-allowed-values-event-outcome.html[ECS outcome field] for allowed values.
```
